### PR TITLE
removing hardcoded vars

### PR DESCRIPTION
**Pull request by clnperez**
adding the ability to pass in network information via install-config.yaml

Signed-off-by: Christy Norman <christy@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by clnperez** [comment]
I need to do the same changes I did for the master configs with the worker configs.

**Comment by clnperez** [comment]
Can you take a look @mjturek, especially at the TF vars changes?

FYI @mkumatag

File: data/data/powervs/topology/pi_network.tf [hunk 1]
```diff
@@ -0,0 +1,37 @@
+## Network
+## These are be optional arguments in the install-config (e.g. Platform)
+## so that users may specify them. Have them be "hidden" in that the survey doesn't ask for them
+## unless the OCP leads disagree.
+## And since they're optional, use the count = construct to conditionally create them if the tf
+## vars are unset.
+
```
Comment (mjturek) [inline data/data/powervs/topology/pi_network.tf:7]: As we discussed in slack, maybe add a comment to make it clear that this is for down the road when the terraform support is available

File: pkg/asset/cluster/tfvars.go [hunk 2]
```diff
@@ -661,11 +661,13 @@ func (t *TerraformVariables) Generate(parents asset.Parents) error {
...
-				APIKey:         session.Session.IAMToken,
-				SSHKey:         installConfig.Config.SSHKey,
-				CISInstanceCRN: crn,
+				MasterConfigs:        masterConfigs,
+				PowerVSZone:          session.Session.Zone,
+				APIKey:               session.Session.IAMToken,
+				PowerVSResourceGroup: installConfig.Config.PowerVS.PowerVSResourceGroup,
+				CISInstanceCRN:       crn,
+				VPCSubnetName:        installConfig.Config.PowerVS.Subnets[0],
+				VPCID:                installConfig.Config.PowerVS.VPC,
```
Comment (mjturek) [inline pkg/asset/cluster/tfvars.go:670]: is `SSHKey` provided somewhere else now? It's required for the terraform to succeed

**Review by mjturek — COMMENTED** [review]
LGTM!

File: pkg/asset/cluster/tfvars.go (hunk 2 above)
Comment (mjturek) [reply pkg/asset/cluster/tfvars.go:670]: Never mind I see it now! It's provided through the master config

**Pull request merged**